samza-azure/src/main/java/org/apache/samza/system/azureblob/AzureBlobConfig.java

```diff
@@ -0,0 +1,189 @@
+
+  // system Level Properties.
+  // fully qualified class name of the AzureBlobWriter impl for the producer system
+  public static final String SYSTEM_WRITER_FACTORY_CLASS_NAME = "systems.%s.producer.azureblob.writer.factory.class";
+  public static final String SYSTEM_WRITER_FACTORY_CLASS_NAME_DEFAULT = "org.apache.samza.system.azureblob.avro.AzureBlobAvroWriterFactory";
+
+  // Azure Storage Account name under which the Azure container representing this system is.
+  // System name = Azure container name (https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#container-names)
+  public static final String SYSTEM_AZURE_ACCOUNT_NAME = SYSTEM_AZUREBLOB_PREFIX + "account.name";
+
+  // Azure Storage Account key associated with the Azure Storage Account
+  public static final String SYSTEM_AZURE_ACCOUNT_KEY  = SYSTEM_AZUREBLOB_PREFIX + "account.key";
+
+  // Whether to use proxy while connecting to Azure Storage
+  public static final String SYSTEM_AZURE_USE_PROXY  = SYSTEM_AZUREBLOB_PREFIX + "proxy.use";
+  public static final boolean SYSTEM_AZURE_USE_PROXY_DEFAULT = false;
+
+  // name of the host to be used as proxy
+  public static final String SYSTEM_AZURE_PROXY_HOSTNAME = SYSTEM_AZUREBLOB_PREFIX + "proxy.hostname";
+
+  // port in the proxy host to be used
+  public static final String SYSTEM_AZURE_PROXY_PORT = SYSTEM_AZUREBLOB_PREFIX + "proxy.port";
+
+  // type of compression to be used before uploading blocks : “none” or “gzip”
+  public static final String SYSTEM_COMPRESSION_TYPE = SYSTEM_AZUREBLOB_PREFIX + "compression.type";
+  private static final CompressionType SYSTEM_COMPRESSION_TYPE_DEFAULT = CompressionType.NONE;
+
+  // maximum size of uncompressed block in bytes
+  public static final String SYSTEM_MAX_FLUSH_THRESHOLD_SIZE = SYSTEM_AZUREBLOB_PREFIX + "maxFlushThresholdSize";
+  private static final int SYSTEM_MAX_FLUSH_THRESHOLD_SIZE_DEFAULT = 10485760;
+
+  // maximum size of uncompressed blob in bytes
+  public static final String SYSTEM_MAX_BLOB_SIZE = SYSTEM_AZUREBLOB_PREFIX + "maxBlobSize";
+  private static final long SYSTEM_MAX_BLOB_SIZE_DEFAULT = Long.MAX_VALUE; // unlimited
+
+  // maximum number of messages in a blob
+  public static final String SYSTEM_MAX_MESSAGES_PER_BLOB = SYSTEM_AZUREBLOB_PREFIX + "maxMessagesPerBlob";
+  private static final long SYSTEM_MAX_MESSAGES_PER_BLOB_DEFAULT = Long.MAX_VALUE; // unlimited
+
+  // number of threads to asynchronously upload blocks
+  public static final String SYSTEM_THREAD_POOL_COUNT = SYSTEM_AZUREBLOB_PREFIX + "threadPoolCount";
+  private static final int SYSTEM_THREAD_POOL_COUNT_DEFAULT = 2;
+
+  // size of the queue to hold blocks ready to be uploaded by asynchronous threads.
+  // If all threads are busy uploading then blocks are queued and if queue is full then main thread will start uploading
+  // which will block processing of incoming messages
+  // Default - Thread Pool Count * 2
```

Comment:
This comment seems out of scope for this class, since it doesn't actually set a default.